var/spack/repos/builtin/packages/cgns/package.py

```diff
@@ -73,6 +73,10 @@ class Cgns(CMakePackage):
     # https://bugs.gentoo.org/662210
     patch("no-matherr.patch", when="@:3.3.1 +tools")
 
+    # patch for gcc14 due to using internal tk type/function,
+    # see https://github.com/CGNS/CGNS/pull/757
+    patch("gcc14.patch", when="@:4.4.0 %gcc@14:")
```

Comment:
Can we retrieve the patch from some https URL?

Reply:
Yes but they `#include <tk-private/generic/tkInt.h>` while we need `#include <tkInt.h>`.

Reply:
Done